custom_components/moonraker/sensor.py

```diff
@@ -452,3 +498,22 @@ def convert_time(time_s):
     return (
         f"{round(time_s // 3600)}h {round(time_s % 3600 // 60)}m {round(time_s % 60)}s"
     )
+
+
+def calculate_current_layer(data):
+    """Calculate current layer"""
+
+    if (
+        data["status"]["print_stats"]["state"] != PRINTSTATES.PRINTING.value
+        or data["status"]["print_stats"]["filename"] == ""
+    ):
+        return ""
+
+    # layer = (current_z - first_layer_height) / layer_height + 1
+    return (
+        int(
```

Comment:
is there a rounding needed? Especially with the bed mesh, I could see the z position to be slightly high or down.

Reply:
now int probably floor the value

Reply:
The cast to int handle it